airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -0,0 +1,255 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import time
+import math
+import requests
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+
+# from airbyte_cdk.models import SyncMode
+# from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator
+
+
+class ShopifyStream(HttpStream, ABC):
+    url_base = ""
+    primary_key = "id"
+    limit = 250
+
+    def __init__(self, shop: str, api_key: str, api_password: str, api_version: str, **kwargs):
+        super().__init__(**kwargs)
+        self.shop = shop
+        self.api_key = api_key
+        self.api_password = api_password
+        self.api_version = api_version
+        self.since_id = 0
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
```

Comment:
how come we are not using the `rel` header from the response to find the next page? https://shopify.dev/tutorials/make-paginated-requests-to-rest-admin-api#how-it-works

Reply:
You're right, there is the opportunity to overcome complex pagination with this, for now it work as expected, but this defenetely should be easier, so I'll work on in

Reply:
@bazarnov any updates on this comment?

Reply:
No, so far didn't get the point of how effectively to do that along with other parameters like since_id, which is doing the same operation on the data.